app/models/concerns/direct_db_conn.rb

```diff
@@ -0,0 +1,65 @@
+module DirectDBConn
```

Comment:
This is not really being used as a concern. You are calling class methods on it directly. You are also passing `table` to many of the methods when it really should be an instance variable I think.

I recommend you move this into the models root directory and call it `BatchInserter`. Make all the methods instance methods and pass `table` to `initialize`. Then you only need to pass `objects` to the instance methods. 

Also, many of the methods should be private.